File: 219_contains_duplicate_2/containsNearbyDuplicate.go
func containsNearbyDuplicate(nums []int, k int) bool {
	hashMap := map[int]int{}

	for i, val := range nums {
		hashVal, ok := hashMap[val]
		if ok && i-hashVal <= k {
			return true
		}
		hashMap[val] = i
	}

	return false
}

File: 219_contains_duplicate_2/containsNearbyDuplicate.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> hash = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int value = nums[i];
            if (hash.containsKey(value) && i - hash.get(value) <= k) {
                return true;
            }
            hash.put(value, i);
        }

        return false;
    }
}